src/services/TestResultsService.ts

```diff
@@ -140,9 +142,9 @@ export class TestResultsService {
     }
 
     // CVSB-7964: Fields Validation for LEC Test Types
-    const missingFieldsForLecTestType: string | undefined = this.validateLecTestTypeFields(payload);
-    if (missingFieldsForLecTestType) {
-      return Promise.reject(new HTTPError(400, missingFieldsForLecTestType + " not present on LEC test type"));
+    const missingFieldsForLecTestType: string[] = this.validateLecTestTypeFields(payload);
+    if (missingFieldsForLecTestType && missingFieldsForLecTestType.length > 0) {
+      return Promise.reject(new HTTPError(400,  {errors: missingFieldsForLecTestType} ));
```

Comment:
Probably still need an explanatory string of some form. e.g. "Missing the following fields: "

Reply:
Tried to make it similar to Joi errors so that mobile dev doesn't have to make a lot of changes.